### PR TITLE
15553 change graphql sub-queries from functions to types

### PR DESCRIPTION
**Pull request by arthanson**
### Fixes: #15553 

GraphQL filtering doesn't work on functions, so define sub-queries as types.

### CODE REVIEW

File: netbox/dcim/graphql/types.py [hunk 1]
```diff
@@ -120,8 +120,7 @@ class ModularComponentTemplateType(ComponentTemplateType):
 )
 class CableTerminationType(NetBoxObjectType):
 
-    @strawberry_django.field
-    def termination(self) -> List[Annotated[Union[
+    termination: List[Annotated[Union[
```
Comment (jeremystretch) [inline netbox/dcim/graphql/types.py:123]: Should this be a List? It returns a single object. (Same question for the other GFK fields.)

File: netbox/ipam/graphql/types.py [hunk 2]
```diff
@@ -227,21 +196,10 @@ def vlans(self) -> List[Annotated["VLANType", strawberry.lazy('ipam.graphql.type
...
-    def importing_vrfs(self) -> List[Annotated["VRFType", strawberry.lazy('ipam.graphql.types')]]:
-        return self.importing_vrfs.all()
-
-    @strawberry_django.field
-    def importing_l2vpns(self) -> List[Annotated["L2VPNType", strawberry.lazy('vpn.graphql.types')]]:
-        return self.importing_l2vpns.all()
+    exporting_l2vpns: List[Annotated["L2VPNType", strawberry.lazy('vpn.graphql.types')]]
+    exporting_vrfs: List[Annotated["VRFType", strawberry.lazy('ipam.graphql.types')]]
+    importing_vrfs: List[Annotated["VRFType", strawberry.lazy('ipam.graphql.types')]]
+    importing_l2vpns: List[Annotated["L2VPNType", strawberry.lazy('vpn.graphql.types')]]
```
Comment (jeremystretch) [inline netbox/ipam/graphql/types.py:202]:
```suggestion
    importing_l2vpns: List[Annotated["L2VPNType", strawberry.lazy('vpn.graphql.types')]]
    exporting_l2vpns: List[Annotated["L2VPNType", strawberry.lazy('vpn.graphql.types')]]
    importing_vrfs: List[Annotated["VRFType", strawberry.lazy('ipam.graphql.types')]]
    exporting_vrfs: List[Annotated["VRFType", strawberry.lazy('ipam.graphql.types')]]
```

**Review by jeremystretch — CHANGES_REQUESTED** [review]
Could you please add some tests to spot check for the functionality this PR fixes?

File: netbox/dcim/graphql/types.py (hunk 1 above)
Comment (arthanson) [reply netbox/dcim/graphql/types.py:123]: good catch - yes, shouldn't be list.

**Comment by jeremystretch** [comment]
Tests will be added under #15569.

**Pull request merged**